datajoint/expression.py

```diff
@@ -439,6 +439,14 @@ def tail(self, limit=25, **fetch_kwargs):
 
     def __len__(self):
         """:return: number of elements in the result set e.g. ``len(q1)``."""
+        try:
+            if self._left[0]:
```

Comment:
This does not appear optimally factored code since the two queries only differ by their fields.

Reply:
I agree. @jverswijver There looks like there could be a simple condition we can introduce here with the combination of a ternary + string formatting operation. Let's tag up on this together since that will likely be a bit easier to demonstrate.